src/python_testing/TC_DISHM_2_1.py

```diff
@@ -0,0 +1,281 @@
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "TH reads from the DUT the SupportedModes attribute", "Verify that the DUT response contains a list of ModeOptionsStruct entries:"
+                     "Verify that the list has two or more entries"
+                     "Verify that PIXIT.DISHM.MODE_CHANGE_FAIL is one of supported_modes_dut"),
+            TestStep(3, "TH reads from the DUT the CurrentMode attribute", "CurrentMode attribute value is an integer value"
+                     "Select a value which is NOT in supported_modes_dut and save it as invalid_mode_th"),
+            TestStep(4, "TH sends a ChangeToMode command to the DUT with NewMode set to old_current_mode_dut",
+                     "DUT responds contains a ChangeToModeResponse command with a SUCCESS (value 0x00) status response"),
+            TestStep(5, "Manually put the device in a state from which it will FAIL to transition to PIXIT.DISHM.MODE_CHANGE_FAIL"),
+            TestStep(6, "TH reads from the DUT the CurrentMode attribute", "CurrentMode attribute value is an integer value"),
+            TestStep(7, "TH sends a ChangeToMode command to the DUT with NewMode set to PIXIT.DISHM.MODE_CHANGE_FAIL",
+                     "DUT responds contains a ChangeToModeResponse command with Status field is set to GenericFailure(0x02), InvalidInMode(0x03), or in the MfgCodes (0x80 to 0xBF) range and StatusText field has a length between 1 and 64"),
+            TestStep(8, "TH reads from the DUT the CurrentMode attribute",
+                     "CurrentMode attribute value is an integer value and equal to old_current_mode_dut"),
+            TestStep(9, "Manually put the device in a state from which it will SUCCESSFULLY transition to PIXIT.DISHM.MODE_CHANGE_OK"),
+            TestStep(10, "TH reads from the DUT the CurrentMode attribute", "CurrentMode attribute value is an integer value"),
+            TestStep(11, "TH sends a ChangeToMode command to the DUT with NewMode set to PIXIT.DISHM.MODE_CHANGE_OK",
+                     "DUT responds contains a ChangeToModeResponse command with a SUCCESS (value 0x00) status response"),
+            TestStep(12, "TH reads from the DUT the CurrentMode attribute",
+                     "CurrentMode attribute value is an integer value and it is the same as step 11"),
+            TestStep(13, "TH sends a ChangeToMode command to the DUT with NewMode set to invalid_mode_th",
+                     "DUT responds contains a ChangeToModeResponse command with a UnsupportedMode(0x01) status response"),
+            TestStep(14, "TH reads from the DUT the CurrentMode attribute",
+                     "CurrentMode attribute value is an integer value and same as step 12")
+        ]
+        return steps
+
+    def pics_TC_DISHM_2_1(self) -> list[str]:
+        pics = [
+            "DISHM.S"
+        ]
+        return pics
+
+    @async_test_body
```

Comment:
```suggestion
    @run_if_endpoint_matches(has_cluster(Clusters.DishwasherMode))
```

Reply:
This was actually ok, but I'm trying to move tests towards this style so that they enforce that the testers are setting endpoints rather than falling back to defaults. It's probably fine for dishwashers, but for some of the other appliance types there can be multiple endpoints with the same clusters.